pennylane/interfaces/jacobian_products.py

```diff
@@ -549,3 +549,52 @@ def compute_jacobian(self, tapes):
         jacs = self._dev_compute_derivatives(tapes)
         self._jacs_cache[tapes] = jacs
         return jacs
+
+
+class DeviceJacobianProducts(JacobianProductCalculator):
+    """Compute jacobian products using the native device methods.
+
+    Args:
+        device (pennylane.devices.Device): the device for execution and derivatives.
+            Must define both the vjp and jvp.
+        execution_config (pennylane.devices.ExecutionConfig): a datastructure containing the options needed to fully
+           describe the execution.
+
+    >>> dev = qml.device('default.qubit')
+    >>> config = qml.devices.ExecutionConfig(gradient_method="adjoint")
+    >>> jpc = DeviceJacobianProducts(dev, config)
+
```

Comment:
Just an idea to be really explicit about the difference to `DeviceDerivatives`.
```suggestion

    In contrast to ``DeviceDerivatives``, this ``JacobianProductCalculator`` lets the native device method decide
    how to obtain the Jacobian product. In particular, this allows the device to perform JVP/VJP evaluations without
    constructing the full Jacobian, if it is capable of doing so.
```